api/src/GearApi.ts

```diff
@@ -35,18 +30,13 @@ export class GearApi extends ApiPromise {
   public code: GearCode;
 
   constructor(options?: GearApiOptions) {
-    const provider = options?.provider
-      ? options.provider
-      : new WsProvider(options?.providerAddress ?? 'ws://127.0.0.1:9944');
+    const { types, providerAddress, ...remain } = options;
+    const provider = remain?.provider || new WsProvider(providerAddress ?? 'ws://127.0.0.1:9944');
+    const defaultTypes = {
+      ...(types || {}),
+      ...gearTypes,
+    };
```

Comment:
```suggestion
    const defaultTypes = types ? { ...types, ...gearTypes } : gearTypes;
```
imo it's more straightforward this way